### ISSUE TITLE
unzipping to a partition other than `C:/` fails with OSError

### ISSUE THREAD

**Issue by froehlichA** [guzba/zippy#25]
On Windows, unzipping to a partition other than `C:/` fails.

When unzipping, you creating a temporary folder & are writing archive contents to that temporary folder, then moving contents to the final folder. However, [Nim currently has a bug where `moveDir` operations across partitions fail](https://github.com/nim-lang/Nim/issues/18216).

I would suggest either:
- waiting until the issue is fixed
- replacing [`moveDir(tempDir, dest)`](https://github.com/guzba/zippy/blob/a059310cb2348bd4b8b2573b6204af289b8cf16b/src/zippy/ziparchives.nim#L417) with a `copyDir` and `removeDir` operation
- write directly to the destination folder instead of using a temporary dir (is there a reason for that?)

**Comment by timotheecour** [comment]
@froehlichA the issue is too vague, what commands reproduce this issue? and also, please follow the issue template, in particular this needs the nim git hash for context

now that your PR that fixes https://github.com/nim-lang/Nim/issues/18216 got merged, this is a performance rather than correctness issue IIURC.

we should:
* review why we need a temporary dir
* if still needed, that temporary dir should be on the same drive as destination

**Comment by guzba** [comment]
Currently, when extracting, a temporary directory is created and all of the files are written there first. Then this directory is moved to the final directory path.

This is done so that if something fails, there isn't a random half finished export polluting the directory. It will either fully work, or fully fail.

**Comment by timotheecour** [comment]
how about this:
```nim
import std/tempfiles
let tmpDir = createTempDir("zippy", ".tmp", dir = destDir)
try:
  unzip to tmpDir
except OsError as e:
  removeDir(tmpDir)
  raise e
moveDir(tmpDir, destDir)
```

**Comment by froehlichA** [comment]
isn't `std/tempfiles` experimental? And not available in the latest stable Nim release yet?

But the approach is correct, you could just create a `./tmp/` folder or a similar folder of any other name in the destination folder - there are guaranteed to be no conflicts with existing directories since `extractAll` fails in case of an existing destination folder anyway.

**Comment by timotheecour** [comment]
> isn't std/tempfiles experimental? And not available in the latest stable Nim release yet?

the API may be subject to change but it's not going to go away/be removed, in any case you can use:
```nim
when (NimMajor, NimMinor, NimPatch) >= (1,5,1): ...
```
so at least i'll be fixed for 1.5.1 onwards.
better than rolling your own IMO, to avoid all the caveats that this involves
